### PR TITLE
[staging-next] gst_all_1.*,python3Packages.gst-python: follow-up for 1.22.9 -> 1.24.2 bump

### LINKED ISSUE #287253: Update request: gst_all_1.gstreamer: 1.22.8  → 1.24.2
- Package name: gst_all_1.gstreamer
- Latest released version: 1.22.9 (https://gstreamer.freedesktop.org/releases/1.22/#1.22.10)

- Current version on the unstable channel: 1.22.8
- Current version on the stable/release channel: 1.22.8

- [ x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@lilyinstarlight 
-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by lilyinstarlight**
## Description of changes

Major GStreamer bump (<https://gstreamer.freedesktop.org/releases/1.24/>)

Fixes #287253

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
  - [ ] x86_64-linux -> aarch64-linux (cross) (in-progress)
  - [x] x86_64-linux (with `qt5Support` override via `nheko`)
  - [x] x86_64-linux (with `gtkSupport` override via `pitivi`)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lilyinstarlight** [comment]
@ofborg eval
(requesting since eval failed due to unrelated reasons on staging earlier)

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 1]
```diff
@@ -148,6 +147,7 @@ stdenv.mkDerivation rec {
   ]) ++ lib.optionals stdenv.isDarwin [
     Cocoa
   ] ++ lib.optionals stdenv.isLinux [
+    libdrm
```
Comment (amaxine) [inline pkgs/development/libraries/gstreamer/good/default.nix:150]: Doh, I caught myself breaking this in base but not good.

**Review by amaxine — APPROVED** [review]
The gst parts of the diff (and especially fixing my mistakes around darwin) look good.

**Review by jansol — APPROVED** [review]
liblc3 changes lgtm

**Review by vcunat — CHANGES_REQUESTED** [review]
This PR breaks the `hotdoc` build for me.  Same error on x86_64-linux and aarch64-darwin:
```
hotdoc-0.16/lib/python3.11/site-packages/hotdoc/core/tests/test_page.py::TestPage::test_meta_schema - AssertionError: InvalidPageMetadata not raised
```

**Comment by vcunat** [comment]
With the hotdoc update reverted it looked OK, so I took that.  Hopefully this stabilization of everything won't be too rough.

**Pull request merged**